pkg/steps/clusterinstall/template.go

```diff
@@ -110,6 +110,12 @@ objects:
     - name: cluster-profile
       secret:
         secretName: ${JOB_NAME_SAFE}-cluster-profile
+    - name: boskos
+      secret:
+        secretName: boskos-credentials
+        items:
```

Comment:
Why do you need to call out the password?

Reply:
It contains other items as well ... we need only the password part here for this pod

```bash
$ oc get secret -n ci boskos-credentials -o yaml | yq -r '.data | keys[]'
ci.htpasswd
password
username

```

Reply:
I'm concerned that this is about ten times more complicated than every other way that we load secrets into any deployment. Why don't we have one file, maybe JSON or YAML, that has the username and password, add it to a known path and give that to the client library? Right now, this secret mounting is special because:

 - you have to know that there is other data you don't want to expose
 - you have to know which parts to expose as a file
 - you have to know how to expose the username as an environment var

Reply:
I was trying to minimize the number of secrets by aggregating the sensitive information into a single secret if they are kind of related. In this case, it is all about boskos basic auth set up (both server side and client sides). Maybe it is totally unnecessary.

I used env. var. because of multi-occurrence. With the current k8s-API way, that reason is no longer valid and env. vars are removed. And username (ci) is in the src control now and I hope that is OK. Password file is mounted as the unknown path is given to ci-operator.

@stevekuznetsov 

See if the current one is better.